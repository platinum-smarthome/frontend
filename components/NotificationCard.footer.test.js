import React, { Component } from 'react'
import enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'

import NotificationCardFooter from './NotificationCard.footer'
import { View, Text, TouchableHighlight } from 'react-native'

enzyme.configure({ adapter: new Adapter()})

let wrapper

beforeEach(() => {
  wrapper = shallow(<NotificationCardFooter />)
})

describe('<NotificationCardFooter/>', () => {
  it('should render <NotificationCardFooter/>', () => {
    expect(wrapper).toBeDefined()
  })
})

describe('<NotificationCardFooter/> child is rendered', () => {
  it('Should render <View/>', () => {
    expect(wrapper.find(<View/>)).toBeDefined()
    expect(wrapper.find('View').length).toBe(1)  
  })
  it('Should render <TouchableHighlight/>', () => {
    expect(wrapper.find(<TouchableHighlight/>)).toBeDefined()
    expect(wrapper.find('TouchableHighlight').length).toBe(1)  
  })
  it('Should render <Text/>', () => {
    expect(wrapper.find(<Text/>)).toBeDefined()
    expect(wrapper.find('Text').length).toBe(1)  
  })
})

describe('snapshot testing', () => {
  it('snapshot', () => {
    const tree = renderer.create(<NotificationCardFooter/>)
    expect(tree).toMatchSnapshot()
  })
})