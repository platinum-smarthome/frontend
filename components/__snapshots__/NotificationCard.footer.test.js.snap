// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderTopColor": "#d3d3d3",
      "borderTopWidth": 1,
      "justifyContent": "center",
      "marginTop": 8,
      "paddingBottom": 4,
      "paddingTop": 10,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "red",
        "borderBottomColor": "#a80000",
        "borderBottomWidth": 3,
        "borderRadius": 3,
        "borderRightColor": "#a80000",
        "borderRightWidth": 3,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "padding": 8,
        }
      }
    >
       DELETE LOGS 
    </Text>
  </View>
</View>
`;
